src/components/OverlayManager/Cart/ProductList.tsx

```diff
@@ -2,54 +2,93 @@ import * as React from "react";
 import { TaxedMoney } from "@components/containers";
 import { Thumbnail } from "@components/molecules";
 import { LineI } from "../../CartTable/ProductRow";
+import ReactSVG from "react-svg";
+import editIcon from "../../../images/baseline-edit-24px.svg";
+import cancelIcon from "../../../images/cancel-24px.svg";
+
+enum LocalStorageKeys {
+  Cart = "cart",
+}
 
 const ProductList: React.SFC<{
   lines: LineI[];
   remove(variantId: string): void;
   add(variantId: string, quantity: number): void;
   subtract(variantId: string, quantity: number): void;
-}> = ({ lines, remove, add, subtract }) => (
-  <ul className="cart__list">
-    {lines.map(line => {
-      return (
-        <li key={line.id} className="cart__list__item">
-          <Thumbnail source={line.product} />
-          <div className="cart__list__item__details">
-            <p>{line.product.name}</p>
-            <div className="cart__list__item__details__variant">
-              <span>SKU: {`W1230-CYOHH`}</span>
-              <span>Dimension: {`10"w x 12"h x 24"d`}</span>
-            </div>
-            <div className="cart__list__item__details__pricing">
-              <p>
-                <TaxedMoney taxedMoney={line.pricing.price} />
-              </p>
-              <QauntSelect
-                id={line.id}
-                quantity={line.quantity}
-                add={add}
-                subtract={subtract}
+}> = ({ lines, remove, add, subtract }) => {
+  const [stockQuantity, setStockQuantity] = React.useState({
+    id: null,
+    stock: 0,
+  });
+  return (
+    <div>
+      <div className="cart__header">
+        <span>Colored Parrot Cushion</span>
+        <ReactSVG path={editIcon} className="cart__edit cart__icon" />
+      </div>
+      <ul className="cart__list">
+        {lines.map(line => {
+          const cart =
+            JSON.parse(localStorage.getItem(LocalStorageKeys.Cart)) || [];
+          const cartItem = cart.find(item => item.variantId === line.id);
+          const cartQuantity = (cartItem && cartItem.quantity) || 0;
+          if (stockQuantity.id !== line.id) {
+            setStockQuantity({
+              id: line.id,
+              stock: line.stockQuantity - cartQuantity,
+            });
+          }
```

Comment:
I'm not sure what you're doing here, but it doesn't look right. Overcomplicated.

You don't need `localStorage` for this.

You just need to compare the item quantity with the backends' stock number. It's really simple.